lib/qstash/http_client.rb

```diff
@@ -36,6 +36,7 @@ def post(body, headers = {})
     def make_request(request)
       request["Authorization"] = "Bearer #{QStash.config.token}"
       request["User-Agent"] = "qstash-ruby/#{QStash::VERSION}"
+      request["Content-Type"] = "application/json"
```

Comment:
The **Content-Type** header is required for URLGroup upsert.

Reply:
Can you share docs as to where this is "required"? ty

As of right now - you can pass the "content-type" in your request within the header object whether you're publishing a message to a url group or creating/deleting url groups.

Reply:
It's not documented. However, all the CURL examples that POST data have this header in the Qstash API documentation.

The response I got with this error was meaningless: "Key: 'AddEndpointsBody.Endpoints' Error:Field validation for 'Endpoints' failed on the 'gt' tag", it was fixed with this header (Upstash opened an internal ticket to fix the error message). I had to debug almost all this gem to figure out what was the error and fix it.

I have attached that conversation: 
[Upstash Nov 18 2024.txt](https://github.com/user-attachments/files/18150758/Upstash.Nov.18.2024.txt)

One thing more, as the gem is sending JSON data in the [code](https://github.com/dvmonroe/qstash-ruby/blob/main/lib/qstash/http_client.rb#L28C22-L28C34), it needs to add the add this header.